controllers/machinedeletionremediation_controller.go

```diff
@@ -462,76 +483,118 @@ func (r *MachineDeletionRemediationReconciler) isTimedOutByNHC(remediation *v1al
 	return false
 }
 
-func getMachineNameNsFromNode(node *v1.Node) (string, string, error) {
-	var nodeAnnotations map[string]string
-	if nodeAnnotations = node.Annotations; nodeAnnotations == nil {
-		return "", "", fmt.Errorf(noAnnotationsError, node.Name)
+// isExpectedNodesNumberRestored checks if the number of nodes associated to the MachineSet is equal to the
+// MachineSet's Replicas value
+func (r *MachineDeletionRemediationReconciler) isExpectedNodesNumberRestored(remediation *v1alpha1.MachineDeletionRemediation) (bool, error) {
+	_, namespace, owner, err := getMachineDataFromRemediation(remediation)
+	if err != nil {
+		return false, errors.Wrap(unrecoverableError, err.Error())
 	}
 
-	var machineNameNs string
-	if machineNameNs = nodeAnnotations[machineAnnotationOpenshift]; len(machineNameNs) == 0 {
-		return "", "", fmt.Errorf(noMachineAnnotationError, node.Name)
+	var machineSet *v1beta1.MachineSet
+	if machineSet, err = r.getMachineSet(owner, namespace); err != nil {
+		r.Log.Error(err, "could not get MachineSet", "machineSet", owner, "namespace", namespace)
+		return false, err
 	}
 
-	if slice := strings.Split(machineNameNs, "/"); len(slice) == 2 {
-		return slice[1], slice[0], nil
+	var nodes *v1.NodeList
+	if nodes, err = r.getMachineSetNodes(machineSet.Name); err != nil {
+		r.Log.Error(err, "could not get MachineSet's nodes", "machineSet", machineSet.Name)
+		return false, err
 	}
-	return "", "", fmt.Errorf(invalidValueMachineAnnotationError, node.Name)
-}
 
-func extractNameAndNamespace(nameNs string, nodeName string) (string, string, error) {
-	if nameNsSlice := strings.Split(nameNs, "/"); len(nameNsSlice) == 2 {
-		return nameNsSlice[1], nameNsSlice[0], nil
+	if machineSet.Spec.Replicas == nil {
```

Comment:
Isn't Replicas will always be set, and by default it is one (// +kubebuilder:default=1)?

Reply:
honestly I don't know, for this reason I was trying to catch any case

Reply:
honestly I don't know, for this reason I was trying to catch any case